### ISSUE TITLE
FR: Support for dataview fields

### ISSUE THREAD

**Issue by isle9** [mProjectsCode/obsidian-meta-bind-plugin#9]
Are there any plans to support dataview fields?

**Comment by mProjectsCode** [comment]
I personally haven't looked into data view fields, since I don't use them, but I can put it on the road map.

**Comment by isle9** [comment]
Any updates on this? Have you checked the dataview API?

**Comment by mProjectsCode** [comment]
not yet, there are currently other things I am working on (templates and the docs)

**Comment by Rhydderch** [comment]
I would love to have this feature as well. (and congrats on this plugin; it's something I've always wanted=)

**Comment by mProjectsCode** [comment]
I am currently working on a library for plugin devs, that provides utility for editing metadata. Dataview fields are on the roadmap for that. 
Once that library is finished I will update this plugin to use it.

**Comment by Rhydderch** [comment]
Awesome. Thank you.

**Comment by mayurankv** [comment]
Is there any update on that? (Also can't believe I'm only just discovering this plugin!).

**Comment by mProjectsCode** [comment]
To keep it simple: The plugin will only support dataview inline fields, once dataview makes it's inline field parser and API available for other plugins to use. As, for a multitude of reasons, I don't want to write my own dataview inline field parser.

**Comment by mayurankv** [comment]
Of course! Fingers crossed on this from dv's side then.

**Comment by alexey-zaharchenko** [comment]
hi, the Dataview inline fields API is indirectly available through a plugin called [MetaEdit](https://github.com/chhoumann/MetaEdit). It's one of the ["Friends of Dataview"](https://blacksmithgu.github.io/obsidian-dataview/friends/) plugins, afaik it used by all other plugins that require access to Dataview fields.

**Comment by mProjectsCode** [comment]
I don't want to rely on a third plugin. 
I have my hopes up that datacore exposes an API for inline fields.

**Comment by dxcore35** [comment]
Hello it is May 2024, the dataview support will be amazing!

**Comment by AveaCallister** [comment]
I thought this was related to this issue. It seems that callouts, inline DQL and Metabind code are incompatible when adjacent to each other. Metabind doesn't detect the frontmatter field being referred to. i.e:

```
> [!note]
>`INPUT[number:field0]`
> [field1:: value] units
```

Results in the input field being empty.

**Comment by mProjectsCode** [comment]
> I thought this was related to this issue. It seems that callouts, inline DQL and Metabind code are incompatible when adjacent to each other. Metabind doesn't detect the frontmatter field being referred to. i.e:
> 
> ```
> > [!note]
> >`INPUT[number:field0]`
> > [field1:: value] units
> ```
> 
> Results in the input field being empty.

Can you make a bug report with a reproducible example?

**Comment by mProjectsCode** [comment]
> Hello it is May 2024, the dataview support will be amazing! Is this implemented? [#107 (comment)](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/issues/107#issuecomment-1731902290)

Dataview still does not provide an API for inline fields afaik.

The issue you linked to is implemented. See the API reference in the docs, at the very bottom of the sidebar.

**Comment by holroy** [comment]
> > Hello it is May 2024, the dataview support will be amazing! Is this implemented? [#107 (comment)](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/issues/107#issuecomment-1731902290)
> 
> Dataview still does not provide an API for inline fields afaik.

Exactly what is it you want to use such an API for? Do you want to change inline fields, similar to properties? Or something else?

Have there been posted an issue in the _Dataview_ repository regarding this?

**Comment by mProjectsCode** [comment]
> > > Hello it is May 2024, the dataview support will be amazing! Is this implemented? [#107 (comment)](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/issues/107#issuecomment-1731902290)
> > 
> > 
> > Dataview still does not provide an API for inline fields afaik.
> 
> Exactly what is it you want to use such an API for? Do you want to change inline fields, similar to properties? Or something else?
> 
> Have there been posted an issue in the _Dataview_ repository regarding this?

Yeah, Dataview inline metadata would need to have an API similar to the Obsidian frontmatter API for me to add Dataview inline metadata support to this plugin.

Regarding your second question: No, I have not. I frankly don't use Dataview much and don't have a lot of personal interest in this FR. I kind of left this open for someone else with more knowlege and interest in Dataview to get such an API for Dataview requested, drafted, and implemented.
